### PR TITLE
#161966909 Implement jwt Authentication

### PR DESCRIPTION
**Pull request by rkemmy**
### What does this PR do?
Implement jwt authentication

### Description of Task to be completed?
* Implement token authentication using jwt at user registration
* Extend token expiry time

### How should this be manually tested?
* `git clone -b ft-jwt-authentication-161966909 https://github.com/andela/ah-infinity-stones and $ cd ah-infinity-stones`
* Run `virtualenv -p python3 venv` to create a virtual environment and `source venv/bin/activate` to activate it
* `pip3 install -r requirements.txt` to install the dependencies
Setup environment variables. Copy all from below and paste them into the .env file. Update the values to your choice.
```export DATABASE_USER='<postgres_username'
export DATABASE_PASSWORD='<postgres_password>'
export HOST='localhost'
export PORT='5432'
export DATABASE_NAME='<database_name>'
```
Activate the env variables by running `source .env`.
5. Run migrations with `python3 manage.py makemigrations` then `python3 manage.py migrate`
6. Run app using  `python3 manage.py runserver`
7. Acess `http://127.0.0.1:8000/api/users` through Postman, provide inputs in the following format and submit.

{ "user": {
"username": "remmy",
"email": "remmy@test.com",
"password": "Password@123"	
}
}

### Screenshot
screenshot 2018-12-13 at 13 31 51

### CODE REVIEW

**Comment by r-wambui** [comment]
The name of this PR is confusing. Also what is the difference of this PR and what Justin did?

**Comment by rkemmy** [comment]
Sorry about the title. I have fixed it. Thanks for pointing it out. The code is similar to Justin's because he integrated jwt for registration before I created a PR and I need to complete the story.

File: authors/apps/authentication/views.py [hunk 1]
```diff
@@ -1,15 +1,18 @@
 from rest_framework import status
-from rest_framework.generics import RetrieveUpdateDestroyAPIView
-from rest_framework.permissions import AllowAny, IsAuthenticated
+from rest_framework.generics import RetrieveUpdateDestroyAPIView, RetrieveUpdateAPIView
```
Comment (codeplus254) [inline authors/apps/authentication/views.py:2]: RetrieveUpdateDestroyAPIView, I believe you're not using this.

Comment (leonardnjura) [reply authors/apps/authentication/views.py:2]: You have a linting eye @codeplus254

Comment (rkemmy) [reply authors/apps/authentication/views.py]: I am using it on the code I am working on. I pushed that code so that you could have something to work on

Comment (rkemmy) [reply authors/apps/authentication/views.py:2]: I rebased @ThaDeveloper's work. He was the one working on views

**Pull request closed without merging**